draft-wang-ppm-dap-taskprov.md

```diff
@@ -231,18 +235,36 @@ struct {
 } VdafConfig;
 ~~~
 
-An extension of this draft may define additional VDAF codepoints in `VdafType`,
-but if an Aggregator doesn't recognize a VDAF codepoint, it MUST opt out of the
-task.
+The lengh prefix of the `vdaf_config` ensures that the `VdafConfig` structure
+can be decoded even if an unrecognized variant is encountered (i.e., an
+unimplemented VDAF).
 
-Apart from the VDAF-specific parameters, this structure includes an opaque
-field `dp_config` to encode differential privacy (DP) parameters. This draft
-doesn't mandate the underlying structure for this field yet.
+Apart from the VDAF-specific parameters, this structure includes a mechanism for
+differential privacy (DP). The opaaque `dp_config` contains the following structure:
+
+~~~
+enum {
+    reserved(0), /* Reserved for testing purposes */
+    none(1),
+    (255)
+} DpMechanism;
+
+struct {
+    DpMechanism dp_mechanism;
+    select (DpConfig.dp_mechanism) {
+        case none: Empty;
+    };
+} DpConfig;
+~~~
 
 > OPEN ISSUE: Should spell out definition of `DpConfig` for various differential
 > privacy mechanisms and parameters. See draft
 > [draft](https://github.com/wangshan/draft-wang-ppm-differential-privacy) for discussion.
 
+The lengh prefix of the `dp_config` ensures that the `DpConfig` structure can
```

Comment:
So now we will have an opaque `vdaf_config`, and another opaque `dp_config` in the decoded `vdaf_config`. I'm not objected to it, but I just want to make sure this is intentional.

Reply:
That's right.

It's a little awkward I think. It might make sense to move `dp_config` out of `VdafConfig` into `TaskConfig`. However we discussed this before and I think @wangshan wants the DP config to stay where it is.